server/src/main/java/org/elasticsearch/common/util/concurrent/ThreadContext.java

```diff
@@ -98,13 +107,23 @@ public ThreadContext(Settings settings) {
             this.defaultHeader = Collections.unmodifiableMap(defaultHeader);
         }
         threadLocal = new ContextThreadLocal();
+        maxWrnHeaderCount = SETTING_HTTP_MAX_WARNING_HEADER_COUNT.get(settings);
+        maxWrnHeaderSize = SETTING_HTTP_MAX_WARNING_HEADER_SIZE.get(settings).getBytes();
     }
 
     @Override
     public void close() throws IOException {
         threadLocal.close();
     }
 
+    public static void setMaxWarningHeaderCount(int newMaxWrnHeaderCount){
+        maxWrnHeaderCount = newMaxWrnHeaderCount;
+    }
+
+    public static void setMaxWarningHeaderSize(ByteSizeValue newMaxWarningHeaderSize){
```

Comment:
Make the method parameter `final` as a guard against accidentally assigning to it instead of the to the member field.